client/Containers/FilterSortControl.jsx

```diff
@@ -0,0 +1,188 @@
+import { useHistory, useLocation } from 'react-router-dom';
+import styled from 'styled-components';
+import { resetPagination } from '../store/product/productPagination';
+import {
+  setProductFilters,
+  filterByType,
+  filterByStyle,
+  filterByRoom,
+  clearFilters,
+} from '../store/product/productFilters';
+import Button from '../components/styles/Button';
+import buildProductQuery from '../components/utils/buildProductQuery';
+import { fetchProductList } from '../store/product/productList';
+
+// const mapStateToProps = ({ productFilters }) => ({ productFilters });
+
+// const mapDispatchToProps = (dispatch) => ({
+//   filter: (option) => {
+//     const { type, style, room, clear } = option;
+//     if (typeof type === 'string') {
+//       dispatch(filterByType(type));
+//     }
+//     if (typeof style === 'string') {
+//       dispatch(filterByStyle(style));
+//     }
+//     if (typeof room === 'string') {
+//       dispatch(filterByRoom(room));
+//     }
+//     if (clear) {
+//       dispatch(clearFilters());
+//     }
+//     dispatch(resetPagination());
+//   },
+// });
+
+const FilterSortBar = styled.div`
+  display: flex;
+  flex-flow: row wrap;
+  justify-content: center;
+  align-items: center;
+  background: black;
+  width: calc(100% - 0.5rem);
+  border-radius: 0 0 0.5rem 0.5rem;
+  color: white;
+  padding: 0 0.5rem 0.5rem 0;
+`;
+
+const AllFilterContainer = styled.div`
+  display: flex;
+  flex-flow: row wrap;
+  justify-content: center;
+  align-items: center;
+`;
+
+const FilterContainer = styled.div`
+  display: flex;
+`;
+
+const FilterLabel = styled.label`
+  padding: 0.5rem 0 0.5rem 0.5rem;
+`;
+
+const FilterSelect = styled.select`
+  margin: 0.5rem;
+`;
+
+const FilterSortControl = () => {
+  const { productFilters, productPagination } = useSelector(
+    ({ productFilters, productPagination }) => ({
+      productFilters,
+      productPagination,
+    })
+  );
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const getProducts = (query) => dispatch(fetchProductList(query));
+  const filter = (options) => {
+    // options: { type?: string, style?: string, room?: string, clear?: true|false }
+    for (const key in productFilters) {
+      if (typeof options[key] === 'string') productFilters[key] = options[key];
+    }
+    if (options.clear) {
+      for (const key in productFilters) {
+        productFilters[key] = '';
+      }
+    }
+    const query = buildProductQuery({
+      productFilters,
+      productPagination: { ...productPagination, page: 1 },
+    });
+    history.push(`/products${query}`);
+    getProducts(query);
+  };
+  const paginate = (options) => {
+    // options: { page?: INT, size?: INT, sort?: STRING }
+    for (const key in productPagination) {
+      if (options[key]) productPagination[key] = options[key];
+    }
+    const query = buildProductQuery({ productFilters, productPagination });
+    history.push(`/products${query}`);
+    getProducts(query);
+  };
```

Comment:
`filter` and `paginate` are functions for manipulating the page view that we attach to the various controls via `onChange` and `onClick`. Note that these do not update our store here; instead we merely push onto `history` an updated `query`, meanwhile relying on the `AllProducts` component to update state when it detects a changed URL.

Reply:
Also realizing that because `componentDidUpdate` will handle it, I do not need the `getProducts` method here at all. Or `dispatch` for that matter.